### PR TITLE
update_config() updates

### PR DESCRIPTION
**Pull request by ronuchit**
* don't need to pass in duplicated env/approach/seed
* experiment_id is optional

### CODE REVIEW

File: src/utils.py [hunk 1]
```diff
@@ -1184,15 +1184,22 @@ def update_config(args: Dict[str, Any], default_seed: int = 123) -> None:
     for k in args:
         if k not in allowed_args:
             raise ValueError(f"Unrecognized arg: {k}")
+    for k in ("env", "approach", "seed"):
+        if not args.get(k, None) and hasattr(CFG, k):
+            # For env, approach, and seed, if we don't pass in a value
```
Comment (ronuchit) [inline src/utils.py:1189]: only want to do this for env/approach/seed bc for instance if we set some boolean flag to False, and then in a later test don't pass anything in, we don't want it to retain False value

File: src/settings.py [hunk 2]
```diff
@@ -166,7 +162,7 @@ def get_arg_specific_settings(args: Dict[str, Any]) -> Dict[str, Any]:
                 {
                     # In the playroom domain, HFF works better.
                     "playroom": "hff",
-                })[args["env"]],
+                })[args.get("env", "")],
```
Comment (ronuchit) [inline src/settings.py:165]: annoying that we can't get rid of this altogether, it's just here to make single unit tests work (both unit test suite and regular execution will always have env and approach set)

File: src/utils.py (hunk 1 above)
Comment (tomsilver) [reply src/utils.py:1189]: isn't `experiment_id` another one of this kind?

**Review by tomsilver — APPROVED** [review]
good job threading this needle, one question

File: src/utils.py (hunk 1 above)
Comment (ronuchit) [reply src/utils.py]: fixed

**Pull request merged**